### PR TITLE
fix specs and remove 'dead' code (Dialyxir)

### PR DESCRIPTION
**Pull request by manuel-rubio**
Hey! I was checking Dialyxir and I found some errors, I had to remove some implementations of builder because during the code they are never gonna be in use. Of course, they could be implemented by the users if they need them. And it's passing correctly now Dialyxir!

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -34,7 +34,8 @@ defmodule Saxy.MixProject do
   defp deps() do
     [
       {:ex_doc, "~> 0.16", only: :dev, runtime: false},
-      {:stream_data, "~> 0.5", only: [:dev, :test]}
+      {:stream_data, "~> 0.5", only: [:dev, :test]},
+      {:dialyxir, "~> 1.0", only: [:dev], runtime: false}
```
Comment (qcam) [inline mix.exs:38]: Thanks for the fix! It looks perfect.

Only small nitpick: I'd prefer not to have `dialyxir` as a dependency at the moment.

Comment (manuel-rubio) [reply mix.exs]: done! :-)

**Comment by qcam** [comment]
@manuel-rubio Thank you so much!

**Comment by manuel-rubio** [comment]
let me know if you publish a new version for hex

**Comment by qcam** [comment]
`1.2.1` released on Hex!

**Comment by qcam** [comment]
Hi @manuel-rubio, due to a bug reported at #74, I have revert the code change in this PR except of the typespec fix. Would you be able to help to check if Dialyxir passes for `master`? Thanks!

**Comment by manuel-rubio** [comment]
@qcam the error is this:
```
lib/saxy/builder.ex:122:callback_type_mismatch
Type mismatch for @callback build/1 in Saxy.Builder behaviour.

Expected type:

  {:cdata, binary()}
  | {:characters, binary()}
  | {:comment, binary()}
  | {:reference, {:decimal, integer()} | {:entity, binary()} | {:hexadecimal, integer()}}
  | {binary(), [{binary(), binary()}], [any()]}


Actual type:
binary()
```
This is meaning that the function `Saxy.Builder.build/1` is never receiving `binary()` so... makes no sense to have those implementations there. Could you provide an example where you are using binary passed into the build function?

**Pull request merged**